cdap-ui/app/cdap/components/PipelineScheduler/ProfilesForSchedule/index.js

```diff
@@ -26,53 +24,48 @@ import classnames from 'classnames';
 import {connect} from 'react-redux';
 import {preventPropagation} from 'services/helpers';
 import StatusMapper from 'services/StatusMapper';
+import ProfilesListView, {extractProfileName} from 'components/PipelineDetails/ProfilesListView';
+
 require('./ProfilesForSchedule.scss');
 
 export const PROFILES_DROPDOWN_DOM_CLASS = 'profiles-list-dropdown';
 
 class ProfilesForSchedule extends Component {
   static propTypes = {
     selectedProfile: PropTypes.string,
-    scheduleStatus: PropTypes.string
+    scheduleStatus: PropTypes.string,
+    profileCustomizations: PropTypes.object
   };
 
+  static defaultProps = {
+    selectedProfile: null,
+    profileCustomizations: {}
+  }
   state = {
-    profiles: null,
     scheduleDetails: null,
-    selectedProfile: this.props.selectedProfile
+    selectedProfile: this.props.selectedProfile,
+    profileCustomizations: this.props.profileCustomizations
   };
 
   componentWillReceiveProps(nextProps) {
     if (nextProps.selectedProfile !== this.state.selectedProfile) {
       this.setState({
-        selectedProfile: nextProps.selectedProfile
+        selectedProfile: nextProps.selectedProfile,
+        profileCustomizations: nextProps.profileCustomizations
       });
     }
   }
 
-  componentDidMount() {
-    this.getProfiles();
-  }
-
-  getProfiles = () => {
-    MyCloudApi.list({
-      namespace: getCurrentNamespace()
-    })
-    .subscribe(
-      profiles => {
-        this.setState({
-          profiles
-        });
-      }
-    );
-  };
-
   selectProfile = (profileName, e) => {
     setSelectedProfile(profileName);
     preventPropagation(e);
   };
 
-  renderProfilesTable = () => {
+  selectProfileWithCustomizations = (profileName, profileCustomizations = {}) => {
```

Comment:
I noticed `setSelectedProfile` already has params `(selectedProfile, profileCustomizations = {})`, so I guess you can just `setSelectedProfile` as the onClick callback directly instead of `selectProfileWithCustomizations`?